marketing_transactional.go

```diff
@@ -0,0 +1,50 @@
+package hubspot
+
+import "fmt"
+
+const transactionalBasePath = "transactional"
+
+// TransactionalService is an interface for the marketing/transactional service of the HubSpot API.
+// Reference: https://developers.hubspot.com/docs/api/marketing/transactional-emails
+type TransactionalService interface {
+	SendSingleEmail(props *SendSingleEmailProperties) (*SendSingleEmailResponse, error)
+}
+
+// Provides the default implementation of TransactionService.
+type TransactionalServiceOp struct {
+	transactionalPath string
+	client            *Client
+}
+
+var _ TransactionalService = (*TransactionalServiceOp)(nil)
+
+type SendSingleEmailMessage struct {
+	To      string   `json:"to"`
+	From    string   `json:"from,omitempty"`
+	SendId  string   `json:"sendId,omitempty"`
+	ReplyTo []string `json:"replyTo,omitempty"`
+	Cc      string   `json:"cc,omitempty"`
+	Bcc     string   `json:"bcc,omitempty"`
+}
+
+type SendSingleEmailProperties struct {
+	EmailId           string                  `json:"emailId"`
+	Message           *SendSingleEmailMessage `json:"message"`
+	ContactProperties *Contact                `json:"contactProperties,omitempty"`
+	CustomProperties  interface{}             `json:"customProperties,omitempty"`
+}
+
+type SendSingleEmailResponse struct {
```

Comment:
Wasn't the Response in the following format?
https://developers.hubspot.com/docs/api/marketing/transactional-emails#endpoint?spec=POST-/marketing/v3/transactional/single-email/send
```
{
  "statusId": "string",
  "sendResult": "SENT",
  "requestedAt": "2023-02-01T02:26:03.209Z",
  "startedAt": "2023-02-01T02:26:03.209Z",
  "completedAt": "2023-02-01T02:26:03.209Z",
  "status": "PENDING",
  "eventId": {
    "created": "2023-02-01T02:26:03.209Z",
    "id": "3fa85f64-5717-4562-b3fc-2c963f66afa6"
  }
}
```

Reply:
Perhaps I'm missing something here - but I orginally tried using `ResponseResource` instead of defining my own, and decoding seemed to produce: 

```
&{ID: Archived:false Associations:<nil> Properties:<nil> CreatedAt:nil UpdatedAt:nil ArchivedAt:nil}
```

Reply:
hmm.. I understood. thank you.